youtube_dl/extractor/bbc.py

```diff
@@ -958,7 +977,31 @@ def _real_extract(self, url):
                             if entry:
                                 self._sort_formats(entry['formats'])
                                 entries.append(entry)
-
+        else:
+            # embed video with playerSettings, eg
+            # https://www.bbc.com/news/av/embed/p07xmg48/50670843
+            settings = self._html_search_regex(
+                r'<script\b[^>]+>.+\.playerSettings\s*=\s*(?P<json>\{.*\})\s*(?:,\s*function\s*\(\s*\)\s*\{\s*["\']use strict.+\(\s*\)\s*)?</script\b',
+                webpage, 'player settings', default='{}', group='json')
+            settings = re.sub(r'([{,]\s*)(\w+)(\s*:)', r'\1"\2"\3', settings)
+            for pling in (('!0', 'true'), ('!1', 'false')):
+                settings = re.sub(r'(:\s*)(%s)(\s*[,}])' % pling[0], r'\1%s\3' % pling[1], settings)
+            settings = self._parse_json(settings, playlist_id, fatal=False)
+            if settings:
+                playlist_object = settings.get('playlistObject', {})
+                if playlist_object:
+                    entry = self._extract_from_playlist_object(playlist_object)
+                    programme_id = entry.get('id')
+                    formats, subtitles = self._download_media_selector(programme_id)
+                    self._sort_formats(formats)
+                    thumbnail = playlist_object.get('holdingImageURL')
+                    entry.update({
+                        'timestamp': timestamp,
+                        'formats': formats,
+                        'subtitles': subtitles,
+                        'thumbnail': thumbnail.replace('$recipe', 'raw') if thumbnail else None,
+                    })
+                    entries.append(entry)
```

Comment:
These looks like candidates for generalization and extracting into a separate method.

Reply:
As above.